addon/-private/system/relationships/state/create.js

```diff
@@ -6,8 +6,13 @@ import EmptyObject from "ember-data/-private/system/empty-object";
 var get = Ember.get;
 
 function createRelationshipFor(record, relationshipMeta, store) {
-  var inverseKey;
-  var inverse = record.type.inverseFor(relationshipMeta.key, store);
+  let inverseKey;
+  let inverse = null;
+  if (!relationshipMeta.options ||
```

Comment:
@fivetanley good point, I'll remove `false`. Personally I think `false` is a better API, but I want to limit the scope of this PR to just the performance improvement.

@krisselden I've updated the PR to use a function as you suggested